tests/org.dataflowanalysis.analysis.tests/src/org/dataflowanalysis/analysis/tests/dfd/MicroSecEndTest.java

```diff
@@ -0,0 +1,248 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.dataflowanalysis.analysis.core.AbstractVertex;
+import org.dataflowanalysis.analysis.dfd.DFDConfidentialityAnalysis;
+import org.dataflowanalysis.analysis.dfd.DFDDataFlowAnalysisBuilder;
+import org.dataflowanalysis.examplemodels.Activator;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import com.google.common.collect.ImmutableMap;
+
+public class MicroSecEndTest {
+    public static final String PROJECT_NAME = "org.dataflowanalysis.examplemodels";
+    public static final String location = Paths.get("casestudies", "TUHH-Models")
+            .toString();
+
+    public static final Map<String, List<Integer>> TUHH_MODELS = ImmutableMap.<String, List<Integer>>builder()
+            .put("anilallewar", List.of(0, 7, 8, 9, 11, 12, 18))
+            .put("apssouza22", List.of(0, 2, 4, 6, 7, 8, 12, 18))
+            .put("callistaenterprise", List.of(0, 2, 4, 6, 11, 18))
+            .put("ewolff", List.of(5, 10, 12, 18))
+            .put("ewolff-kafka", List.of(0, 3, 4, 5, 6, 7, 8, 9, 18))
+            .put("fernandoabcampos", List.of(18))
+            .put("georgwittberger", List.of(0, 2, 3, 4, 5, 6, 7, 8, 10, 11, 12, 18))
+            .put("jferrater", List.of(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 18))
+            .put("koushikkothagal", List.of(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 18))
+            .put("mdeket", List.of(5))
+            .put("mudigal-technologies", List.of(0, 2, 4, 5, 7, 8, 11, 18))
+            .put("rohitghatol", List.of(10, 12, 18))
+            .put("spring-petclinic", List.of(0, 2, 3, 4, 5, 6, 7, 8, 9, 18))
+            .put("sqshq", List.of(0, 7, 8, 9, 10, 11, 12, 18))
+            .put("yidongnan", List.of(0, 2, 3, 4, 5, 6, 7, 8, 9, 18))
+            .build();
+    
+
+    public DFDConfidentialityAnalysis buildAnalysis(String name) {
+        var DataFlowDiagramPath = name + ".dataflowdiagram";
+        var DataDictionaryPath = name + ".datadictionary";
+
+        return new DFDDataFlowAnalysisBuilder().standalone()
+                .modelProjectName(PROJECT_NAME)
+                .usePluginActivator(Activator.class)
+                .useDataFlowDiagram(DataFlowDiagramPath)
+                .useDataDictionary(DataDictionaryPath)
+                .build();
+    }
+
+
+    public void analysis(String model,int variant,Map<Integer, Map<Integer, List<AbstractVertex<?>>>> violationsMap) {
+        var analysis = buildAnalysis(model);
+        analysis.initializeAnalysis();
+        System.out.println(analysis.toString());
+        var flowGraph = analysis.findFlowGraphs();
+        flowGraph.evaluate();
+        
+        assertFalse(flowGraph.getTransposeFlowGraphs()
+                .isEmpty());
+
+        var hasSecretManager = false;
+        var hasLoggingServer = false;
+        for (var aTFG : flowGraph.getTransposeFlowGraphs()) {
```

Comment:
If you want to use the abbreviation here, I would use `tfg` as variable name.
Else, and what is probably more readable, i would use `transposeFlowGraph` or `flowGraph` here

Reply:
changed to transposeFlowGraph

Reply:
also changed query to loop